amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/rm/rest/api/retentionschedule/RetentionScheduleActionRelation.java

```diff
@@ -0,0 +1,225 @@
+                RecordsManagementModel.ASSOC_DISPOSITION_ACTION_DEFINITIONS,
+                QName.createQName(NamespaceService.CONTENT_MODEL_1_0_URI,
+                        QName.createValidLocalName(nodeInfos.get(0).getName())),
+                RecordsManagementModel.TYPE_DISPOSITION_ACTION_DEFINITION, actionDefinitionParams).getChildRef();
+        DispositionSchedule dispositionSchedule = new DispositionScheduleImpl(service, nodeService, retentionScheduleNodeRef);
+        DispositionActionDefinition dispositionActionDefinition = dispositionSchedule.getDispositionActionDefinition(actionNodeRef.getId());
+        List<RetentionScheduleActionDefinition> responseActions = new ArrayList<>();
+        if (dispositionActionDefinition != null)
+        {
+            responseActions.add(nodesModelFactory.mapRetentionScheduleActionDefData(dispositionActionDefinition));
+        }
+        return responseActions;
+    }
+
+    @Override
+    @WebApiDescription(title = "Return a paged list of retention schedule action definition based on the 'retentionScheduleId'")
+    public CollectionWithPagingInfo<RetentionScheduleActionDefinition> readAll(String retentionScheduleId, Parameters parameters)
+    {
+        checkNotBlank("retentionScheduleId", retentionScheduleId);
+        mandatory("parameters", parameters);
+        NodeRef retentionScheduleNodeRef = apiUtils.lookupAndValidateNodeType(retentionScheduleId, RecordsManagementModel.TYPE_DISPOSITION_SCHEDULE);
+        List<DispositionActionDefinition> actions = nodesModelFactory.getRetentionActions(retentionScheduleNodeRef);
+        List<RetentionScheduleActionDefinition> actionDefinitionList = actions.stream()
+                .map(nodesModelFactory::mapRetentionScheduleActionDefData)
+                .collect(Collectors.toList());
+        return CollectionWithPagingInfo.asPaged(parameters.getPaging(), actionDefinitionList, false,
+                actionDefinitionList.size());
+    }
+
+    /**
+     * this method is used to validate the order of the retention schedule step
+     * @param retentionScheduleNodeRef nodeRef
+     * @param retentionScheduleActionDefinition retention schedule action definition
+     */
+    private void retentionScheduleStepValidation(NodeRef retentionScheduleNodeRef, RetentionScheduleActionDefinition retentionScheduleActionDefinition)
+    {
+        List<DispositionActionDefinition> actions = nodesModelFactory.getRetentionActions(retentionScheduleNodeRef);
+        Set<String> completedActions = new HashSet<>();
+        if (!actions.isEmpty())
+        {
+            completedActions = actions.stream()
+                    .map(DispositionActionDefinition::getName)
+                    .collect(Collectors.toSet());
+        }
+
+        if (completedActions.contains(RetentionSteps.DESTROY.stepName))
+        {
+            throw new ConstraintViolatedException("Invalid Step - destroy action is already added . No other action is allowed after Destroy.");
```

Comment:
Will it work with Multi-lingual environment?

Reply:
We have verified other rest api response. We always get the reponse in english locale from rest api.
It's reponsibility of the front end application to display messages in different locales.